src/mod_vcard.erl

```diff
@@ -196,6 +204,114 @@ stop(VHost) ->
     gen_server:call(Proc, stop),
     ejabberd_sup:stop_child(Proc).
 
+%%--------------------------------------------------------------------
+%% config_spec
+%%--------------------------------------------------------------------
+
+-spec config_spec() -> mongoose_config_spec:config_section().
+config_spec() ->
+    #section{
+       items = #{<<"iqdisc">> => mongoose_config_spec:iqdisc(),
+                 <<"host">> => #option{type = string,
+                                       validate = non_empty},
+                 <<"search">> => #option{type = boolean},
+                 <<"backend">> => #option{type = atom,
+                                          validate = {module, mod_vcard}},
+                 <<"matches">> => #option{type = int_or_infinity,
+                                          validate = non_negative},
+                 <<"ldap_pool_tag">> => #option{type = atom,
+                                                validate = non_empty},
```

Comment:
Validate `pool_name` here

Reply:
I took those ldap types and validations from [ldap_auth](https://github.com/esl/MongooseIM/blob/37f000962df086f5eebeb718dc60fc9e41dd43d0/src/config/mongoose_config_spec.erl#L444), maybe we should change there as well?

Reply:
Probably yes, but out of scope here.